java-manta-client/src/main/java/com/joyent/manta/config/ConfigContext.java

```diff
@@ -34,4 +34,33 @@
      * @return General connection timeout for the Manta service.
      */
     Integer getTimeout();
+
+    /**
+     * @return String of home directory based on Manta username.
+     */
+    String getMantaHomeDirectory();
+
+    /**
+     * Extracts the home directory based on the Manta account name.
+     *
+     * @param mantaUser user associated with account
+     * @return the root account holder name prefixed with a slash or null
+     *         upon null mantaUser
+     */
+    static String deriveHomeDirectoryFromUser(final String mantaUser) {
+        if (mantaUser == null) {
+            return null;
+        }
+
+        final String home;
+
+        if (mantaUser.contains("/")) {
```

Comment:
I'd recommend a slightly stronger validation here... maybe check to make sure the manta user config value contains 0 or 1 "/" characters but no more.  And also make sure the "/" is not at the beginning, or the 0-th element is going to result in an empty string (I think).